### PR TITLE
Added config option for previewable blue coins

### PR DESCRIPTION
**Pull request by ColinT**
Adds config option for the following feature:
 - When Mario is standing on an idle blue coin switch, blue coins associated with the switch will become visible.

Compatible with `BLUE_COIN_SWITCH_RETRY` config option.

### CODE REVIEW

**Comment by thecozies** [comment]
This is an awesome feature, we're just in the middle of a big refactor so this will have to wait until afterwards. Thanks for submitting!

**Comment by ColinT** [comment]
Rebased for v2.0.x

**Review by gheskett — APPROVED** [review]
Code-wise, this looks good to me. The main question is whether others agree this define and its execution are suitable enough for the repo. Some aesthetic difference such as flickering or transparency may make this feature more attractive, but those are largely subjective and some ideas could potentially require additional changes that are more complicated.

File: src/game/behaviors/blue_coin.inc.c [hunk 1]
```diff
@@ -55,6 +60,7 @@ void bhv_hidden_blue_coin_loop(void) {
             if (cur_obj_wait_then_blink(200, 20)) {
 #ifdef BLUE_COIN_SWITCH_RETRY
                 o->oAction = HIDDEN_BLUE_COIN_ACT_INACTIVE;
+                o->header.gfx.node.flags &= ~GRAPH_RENDER_INVISIBLE;
```
[the remark this answers is not in the document]
Comment (gheskett) [reply src/game/behaviors/blue_coin.inc.c:63]: I guess the blinking function apparently does, so this change makes sense then.

**Comment by gheskett** [comment]
Moved to #507

**Pull request closed without merging**